File: Ahorn/triggers/WASDMovementTrigger.jl
module FrostHelperWASDMovementTrigger

using ..Ahorn, Maple

@mapdef Trigger "FrostHelper/WASDMovementTrigger" WASDMovementTrigger(x::Integer, y::Integer, width::Integer=16, height::Integer=16, hitboxWidth::Integer=2, speed::Number=80.0, texture::String="util/pixel")

const placements = Ahorn.PlacementDict(
	"WASD Movement (Frost Helper)" => Ahorn.EntityPlacement(
		WASDMovementTrigger,
		"rectangle"
	)
)

end

File: Loenn/triggers/WASDMovementTrigger.lua
local jautils = require("mods").requireFromPlugin("libraries.jautils")

local trigger = {
    name = "FrostHelper/WASDMovementTrigger",
}

jautils.createPlacementsPreserveOrder(trigger, "default", {
    { "hitboxWidth", 2, "integer" },
    { "texture", "util/pixel" },
    { "speed", 80.0 },
})

return trigger